src/Test/L0/Plugin/TestFileShareProvider/TestFileShareProviderL0.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Plugins.PipelineArtifact;
+using Agent.Sdk;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Tests;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Xunit;
+
+namespace Test.L0.Plugin.TestFileShareProviderL0
+{
+    public class TestFileShareProviderL0
+    {
+        private const string TestSourceFolder = "sourceFolder";
+        private const string TestDestFolder = "destFolder";
+        private const string TestDownloadSourceFolder = "sourceDownloadFolder";
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Plugin")]
+        public async Task TestPublishArtifactAsync()
+        {
+            if(!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Assert.True(true);
```

Comment:
>true [](start = 28, length = 4)

What;s the point of this? YOu need to spew a message indicating what's wrong.

Reply:
also you're shortcircuiting

---
In reply to: [319687143](https://github.com/microsoft/azure-pipelines-agent/pull/2406#discussion_r319687143) [](ancestors = 319687143)

Reply:
The test shouldn't run in non windows environment and note agent is using Xunit.

Reply:
^ we need at least a comment indicating this because it looks confusing without one.

Reply:
ok adding it.

Reply:
Thought it's obvious since it's checking the platform first..

Reply:
It would be obvious with an Assert.Inconclusive, but because xUnit doesn't have one we'll need a comment there.

Reply:
Agree. Added the comment.